cpg-analysis/src/test/kotlin/de/fraunhofer/aisec/cpg/analysis/fsm/ComplexDFAOrderEvaluationTest.kt

```diff
@@ -409,6 +411,41 @@ class ComplexDFAOrderEvaluationTest {
         assertFalse(everythingOk, "Expected incorrect order")
     }
 
+    @Test
+    fun testSuccessWhileLoop2FSM() {
+        val functionOk =
+            tu
+                .getDeclarationsByName("ComplexOrder", RecordDeclaration::class.java)
+                .firstOrNull()
+                ?.declarations
+                ?.firstOrNull { d -> d.name == "okWhile2" } as
+                FunctionDeclaration?
+
+        assertNotNull(functionOk)
+
+        val p7Decl = (functionOk.body as CompoundStatement).statements[0] as? DeclarationStatement
```

Comment:
you can use the `bodyOrNull<DeclarationStatement>(index)` call as a shorthand for this